charms/worker/k8s/src/charm.py

```diff
@@ -248,6 +253,39 @@ def _configure_cos_integration(self):
         if relation := self.model.get_relation("cos-tokens"):
             self.collector.request(relation)
 
+    def _configure_datastore(self, config: BootstrapConfig):
+        """Configure the datastore for the Kubernetes cluster.
+
+        Args:
+            config (BootstrapConfig): The bootstrap configuration object for
+                the Kubernetes cluster that is being configured. This object
+                will be modified in-place to include etcd's configuration details.
+        """
+        datastore = self.config.get("datastore")
+
+        if datastore not in SUPPORTED_DATASTORES:
+            log.error(
+                "Invalid datastore: %s. Supported values: %s",
+                datastore,
+                ", ".join(SUPPORTED_DATASTORES),
+            )
```

Comment:
You could have the charm block here waiting for a config change

```suggestion
        if datastore not in SUPPORTED_DATASTORES:
            status.add(ops.BlockedStatus(f"Invalid datastore: {datastore}"))
       assert datastore in SUPPORTED_DATASTORES
```